dotnet/samples/KernelSyntaxExamples/Example00_01_PlayFabDataQnA.cs

```diff
@@ -167,85 +179,116 @@ private static async Task<IKernel> GetKernelAsync()
             .Build();
 
         // We're using volotile memory, so pre-load it with data
-        await InitializeMemoryAsync(kernel);
+        await InitializeMemoryAsync(kernel, TestConfiguration.PlayFab.TitleId, cancellationToken);
 
         return kernel;
     }
 
-    private static async Task InitializeMemoryAsync(IKernel kernel)
+    private static async Task InitializeMemoryAsync(IKernel kernel, string titleId, CancellationToken cancellationToken)
     {
-        DateTime today = DateTime.UtcNow;
+        if (AllTitleReports == null)
+        {
+            DateTime today = DateTime.UtcNow.Date;
+
+            using var reportDataFetcher = new ReportDataFetcher(
+                titleId,
+                TestConfiguration.PlayFab.ReportsCosmosDBEndpoint,
+                TestConfiguration.PlayFab.ReportsCosmosDBKey,
+                TestConfiguration.PlayFab.ReportsCosmosDBDatabaseName,
+                TestConfiguration.PlayFab.ReportsCosmosDBContainerName);
+
+            IList<GameReport> gameReports = await reportDataFetcher.FetchByQueryAsync(
+                $"SELECT * FROM c WHERE c.TitleId='{titleId}' and c.ReportDate>='{today.AddDays(-3):yyyy-MM-dd}'",
+                cancellationToken);
+
+            Dictionary<string, GameReport> latestReports = gameReports
+              .GroupBy(report => report.ReportName)
+              .Select(group => group.OrderByDescending(report => report.ReportDate).First())
+              .ToDictionary(report => report.ReportName);
+
+            if (!latestReports.ContainsKey("EngagementMetricsRollupReportCSV"))
+            {
+                GameReport gameEngagementRollupReport = (await reportDataFetcher.FetchByQueryAsync(
+                     $"SELECT TOP 1 * FROM c WHERE c.TitleId='{titleId}' and c.ReportName='EngagementMetricsRollupReportCSV' ORDER BY c.ReportDate DESC",
```

Comment:
The below query data already should be part of the previous query. Not sure making one more query will help anything.

$"SELECT TOP 1 * FROM c WHERE c.TitleId='{titleId}'

Reply:
I thought this one always has the ReportDate set to the first day of the month. Checking...

Reply:
Confirmed this is needed, since this report data is always the first date of the month